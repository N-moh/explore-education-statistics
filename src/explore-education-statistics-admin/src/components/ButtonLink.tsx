import classNames from 'classnames';
import React, { ReactNode } from 'react';
import { Link as RouterLink, LinkProps } from 'react-router-dom';

import { SetRealButtonLink } from '@common/components/ButtonLink';

type Props = {
  children: ReactNode;
  className?: string;
  disabled?: boolean;
  testId?: string;
} & LinkProps;

const ButtonLink = ({
  children,
  className,
  disabled = false,
  to,
  testId,
  ...props
}: Props) => {
  return (
    <RouterLink
      {...props}
      to={to}
      className={classNames(
        'govuk-button',
        {
          'govuk-button--disabled': disabled,
        },
        className,
      )}
      role="button"
      aria-disabled={disabled}
      data-testid={testId}
    >
      {children}
    </RouterLink>
  );
};


SetRealButtonLink(ButtonLink);

export default ButtonLink;
