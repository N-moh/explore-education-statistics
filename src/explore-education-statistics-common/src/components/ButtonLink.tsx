import React, { ReactNode } from 'react';

export interface ButtonLinkProps {
  children: ReactNode;
  as: string;
  href: string;
  [key:string]: any;
}

let RealButtonLink : React.ComponentType<ButtonLinkProps>;
export type ButtonLinkType = typeof RealButtonLink;

export const SetRealButtonLink = (buttonLink : ButtonLinkType) => {
  RealButtonLink = buttonLink;
  return RealButtonLink;
};

const ButtonLink = (props: ButtonLinkProps) => {
  return (
    <RealButtonLink {...props} />
  );
};

export default ButtonLink;