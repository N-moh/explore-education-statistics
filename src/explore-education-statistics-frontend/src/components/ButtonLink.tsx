import classNames from 'classnames';
import Link from 'next/link';
import React from 'react';
import { LinkProps } from './Link';

type Props = {
  disabled?: boolean;
} & LinkProps;

const ButtonLink = ({
  as,
  children,
  className,
  disabled = false,
  to,
  href,
  prefetch,
  ...props
}: Props) => {
  return (
    <Link href={href || to} as={as} prefetch={prefetch}>
      <a
        {...props}
        className={classNames(
          'govuk-button',
          {
            'govuk-button--disabled': disabled,
          },
          className,
        )}
        role="button"
        aria-disabled={disabled}
      >
        {children}
      </a>
    </Link>
  );
};

export default ButtonLink;
