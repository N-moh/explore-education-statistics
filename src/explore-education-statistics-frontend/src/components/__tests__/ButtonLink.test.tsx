import React from 'react';
import { render } from 'react-testing-library';
import ButtonLink from '../ButtonLink';

describe('ButtonLink', () => {
  test('renders correctly with required props', () => {
    const { container } = render(<ButtonLink to="/">Test button</ButtonLink>);

    expect(container.innerHTML).toMatchSnapshot();
  });

  test('correctly set disabled attributes', () => {
    const { getByText } = render(<ButtonLink to="/">Test button</ButtonLink>);

    const button = getByText('Test button');

    expect(button).not.toHaveAttribute('disabled');
    expect(button).toHaveAttribute('aria-disabled', 'false');
  });
});
