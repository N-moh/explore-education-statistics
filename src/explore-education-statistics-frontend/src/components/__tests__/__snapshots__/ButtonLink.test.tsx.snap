// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink renders correctly with required props 1`] = `
<a href="/"
   class="govuk-button"
   role="button"
   aria-disabled="false"
>
  Test button
</a>
`;
